pkg/controller/deployment/util/deployment_util.go

```diff
@@ -861,6 +861,19 @@ func DeploymentTimedOut(deployment *extensions.Deployment, newStatus *extensions
 	if condition == nil {
 		return false
 	}
+	// If the previous condition has been a successful rollout then we shouldn't try to
+	// estimate any progress. Scenario:
+	//
+	// * progressDeadlineSeconds is smaller than the difference between now and the time
+	//   the last rollout finished in the past.
+	// * the creation of a new ReplicaSet triggers a resync of the Deployment prior to the
+	//   cached copy of the Deployment getting updated with the status.condition that indicates
+	//   the creation of the new ReplicaSet.
+	//
+	// In that case, we are going to falsely indicate that the Deployment has failed syncing.
```

Comment:
The last line of this comment is a little unclear

Reply:
Thanks, now that I am reading it again, indeed it is. Updated.